go/docs/port_detection.md

```diff
@@ -0,0 +1,299 @@
+## Port Detection in Alizer
+
+Port detection is one of the step included during component detection and it refers to the ports used by the component that should be opened in the container.
+Because of the different nature of frameworks supported, Alizer tries to use customized ways to detect ports from source code, if necessary.
+Only ports with value > 0 and < 65535 are valid.
+
+There are three detection strategies currently available:
+1) Docker file - Alizer looks for a Dockerfile in the root folder and tries to extract ports from it.
+2) Compose file - Alizer searches for a docker-compose file in the root folder and tries to extract port of the service from it
+3) Source - If a framework has been detected during component detection, a customized detection is performed. Below a detailed overview of the different strategies for each supported framework.
+
+By default, Alizer perform a docker, compose and source detection, in this order.
+If one or more ports are found in one step, the rest are skipped. (E.g - if a port is found in the docker file, Alizer will not search and analyze 
+any docker-compose file and the source code).
+
+It is also possible to customize the way Alizer searches for ports by defining the detection strategies to use and their order.
+For example, by using a source and docker strategies, Alizer analyzes the source code and then the docker file, if any. Or docker and compose, to avoid analyzing the source.
+
+If no port is found an empty list is returned.
+
+### Port detection with a docker-compose file
+
+The port detection in a `[compose|docker-compose].[yml|yaml]` file targets the `expose` and `ports` fields of the service related to the component, if any. 
+To search for the specific component, Alizer look at the `build` field.
+
+Example of `expose` - the result is [3000,8000]
+```
+services:
+  web:
+    ...
+    expose:
+      - "3000"
+      - "8000"
+  myapp2:
+    ...
+    expose:
+      - "5000"
+```
+
+Example of short syntax `ports` (`[HOST:]CONTAINER[/PROTOCOL]`) - the result is [3000,3005,8000,8081,8002,6060]
+```
+services:
+  web:
+    ...
+  ports:
```

Comment:
Aren't we missing a tab here ? Does the `ports` refer to the `web` service or to `services` ?

Reply:
You're right. Fixed. Thanks!!